src/Database/Schema/SchemaDialect.php

```diff
@@ -112,6 +114,56 @@ protected function _convertConstraintColumns($references): string
         ));
     }
 
+    /**
+     * Tries to use a matching database type to generate the SQL
+     * fragment for a single column in a table.
+     *
+     * @param string $columnType The column type.
+     * @param \Cake\Database\Schema\TableSchemaInterface $schema The table schema instance the column is in.
+     * @param string $column The name of the column.
+     * @return string|null An SQL fragment, or `null` in case no corresponding type was found or the type didn't provide
+     *  custom column SQL.
+     */
+    protected function _getTypeSpecificColumnSql(
+        string $columnType,
+        TableSchemaInterface $schema,
+        string $column
+    ): ?string {
+        if (!TypeFactory::getMap($columnType)) {
+            return null;
+        }
+
+        $type = TypeFactory::build($columnType);
```

Comment:
I was working on update the docs to be more clear on what will be inside `$definition`, and this line made me think. How is this supposed to work with custom types? Is the goal to only handle custom types that overwrite an existing 'basic' type, or handles a column type that the ORM cannot currently handle?

Reply:
Both. The latter was kinda shown before my change to simplify the tests, problem was that generating `TEXT` in the type would cause the made up `schemaawaretype` type to be lost between `getColumnSql()` and `convertColumnDefinition()`, so I've chosen to remap the existing type.

I could add an additional driver specific type, like for example `POINT`, and use that for MySQL/Postgres-only integration tests that show handling types unknown to CakePHP? Maybe I could even get it to work with SQL Server, but that requires a different column type IIRC, `GEOGRAPHY` I think.

Reply:
I'm not sure the test needs to cover the a DB specific type as it complicates maintenance.